aspnetcore/tutorials/grpc/grpc-client.md

```diff
@@ -0,0 +1,171 @@
+---
+title: "Tutorial: Create a .NET Core gRPC client"
+author: juntaoluo
+description: This series of tutorials shows how to create a gRPC Service on ASP.NET Core. Learn how to create a gRPC Service project, edit a proto file, and add an duplex streaming call.
+monikerRange: '>= aspnetcore-3.0'
+ms.author: johluo
+ms.date: 4/10/2019
+uid: tutorials/grpc/grpc-client
+---
+# Tutorial: Create a .NET Core gRPC client
+
+By [John Luo](https://github.com/juntaoluo)
+
+This tutorial teaches how to create a .NET Core gRPC client that can communicate with gRPC services.
+
+At the end, you'll have a gRPC client that communicates with the gRPC Greeter service.
+
+[!INCLUDE[View or download sample code](~/includes/grpc/downloadClient.md)]
+
+In this tutorial, you:
+
+> [!div class="checklist"]
+> * Create a gRPC client.
+> * Run the service against the gRPC Greeter service created in the previous tutorial.
+> * Examine the project files.
+
+[!INCLUDE[](~/includes/net-core-prereqs-all-3.0.md)]
+
+## Create a .NET console application
+
+# [Visual Studio](#tab/visual-studio)
+
+Follow the instructions [here](https://docs.microsoft.com/en-us/dotnet/core/tutorials/with-visual-studio) to create a console app with the name GrpcGreeterClient.
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+Follow the instructions [here](https://docs.microsoft.com/en-us/dotnet/core/tutorials/with-visual-studio-code) to create a console app with the name GrpcGreeterClient.
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+Follow the instructions [here](https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-on-mac-vs-full-solution) to create a console app with the name GrpcGreeterClient.
+
+<!-- End of VS tabs -->
+
+---
+
+## Add required packages
+
+The following packages need to be added the gRPC client project:
+
+* [Grpc.Core](https://www.nuget.org/packages/Grpc.Core) which contains the C# API for the C-core client.
+* [Google.Protobuf](https://www.nuget.org/packages/Google.Protobuf/) which contains protobuf message APIs for C#.
+* [Grpc.Tools](https://www.nuget.org/packages/Grpc.Tools/) which contains C# tooling support for protobuf files. The tooling package isn't required at runtime, so the dependency is marked with `PrivateAssets="All"`.
+
+## Add the greet.proto file
+
+Copy the **Protos\greet.proto** file from the gRPC Greeter service to the gRPC client project. Add the **greet.proto** file to the `<Protobuf>` item group of the GrpcGreeterClient project file:
+
+```XML
+<Protobuf Include="Protos\greet.proto" GrpcServices="Client" />
+```
+
+> [!NOTE]
+> You can open the project file of GrpcGreeterClient by right-clicking the project and selecting the **Edit GrpcGreeterClient.csproj** option from the dropdown menu.
+> ![new ASP.NET Core Web Application](grpc-start/_static/edit_csproj.png)
```

Comment:
I think you need extra lines with `>` above and below this image so it appears on its own line. Currently it looks like this:

![image](https://user-images.githubusercontent.com/303201/56170210-c8816580-6034-11e9-93fb-8bcd07466592.png)

Reply:
This will look ugly even when I fix the formatting... but luckily it's only temporary.